src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -0,0 +1,137 @@
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
+import Logger from "bunyan";
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { TokenType, verifyQsh } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
+import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
+import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+
+export const jiraSymmetricJwtMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+
+	if (!await booleanFlag(BooleanFlags.NEW_JWT_VALIDATION, false)) {
+		return next();
+	}
+
+	const token = req.query?.["jwt"] || req.cookies?.["jwt"];
+
+	if (token) {
+
+		let issuer;
+		try {
+			issuer = getIssuer(token, req.log);
+		} catch (err) {
+			req.log.warn({ err }, "Could not get issuer");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const installation = await Installation.getForClientKey(issuer);
+		if (!installation) {
+			req.log.warn("No Installation found");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const secret = await installation.decrypt("encryptedSharedSecret");
+
+		const tokenType = checkPathValidity(req.originalUrl) && req.method == "GET" ? TokenType.normal : TokenType.context;
```

Comment:
do we have a test for at least one of the URL from the list, to make sure we pick the token type correctly?

Reply:
test case added